bread/layout/components/form.py

```diff
@@ -286,71 +281,50 @@ def _mapwidget(
-    # TODO: This can be simplified, and improved
-    if field.field.localize:
-        field.field.widget.is_localized = True
-    attrs = dict(field.field.widget.attrs)
-    attrs.update(widgetattributes)
-    attrs = field.build_widget_attrs(attrs)
-    if getattr(field.field.widget, "allow_multiple_selected", False):
-        attrs["multiple"] = True
-    if field.auto_id and "id" not in field.field.widget.attrs:
-        attrs.setdefault("id", field.html_initial_id if only_initial else field.auto_id)
-    if "name" not in attrs:
-        attrs["name"] = field.html_initial_name if only_initial else field.html_name
-    value = field.field.widget.format_value(field.value())
-    if value is not None and "value" not in attrs:
-        attrs["value"] = value
-
+    widgetattributes = update_widgetattributes(field, only_initial, widgetattributes)
     elementattributes = {
-        **getattr(field.field, "layout_kwargs", {}),
-        **elementattributes,
-    }
-
-    kwargs = {
         "label": field.label,
         "help_text": field.help_text,
         "errors": field.errors,
         "disabled": field.field.disabled,
         "required": field.field.required,
-        "widgetattributes": attrs,
-        **elementattributes,
+        **getattr(field.field, "layout_kwargs", {}),
+        **(elementattributes or {}),
     }
+
     if isinstance(field.field.widget, forms.CheckboxInput):
-        attrs["checked"] = field.value()
+        widgetattributes["checked"] = field.value()
         return hg.DIV(
-            Checkbox(**kwargs),
+            Checkbox(
+                widgetattributes=widgetattributes,
+                **elementattributes,
+            ),
             _class="bx--form-item",
         )
-    if isinstance(field.field.widget, forms.CheckboxSelectMultiple):
-
-        def extract_attrs(data):
-            return {
-                "name": data["name"],
-                "value": data["value"],
-                "checked": data["selected"],
-                **data["attrs"],
-                **widgetattributes,
-            }
 
-        del kwargs["label"]
-        del kwargs["required"]
+    if isinstance(field.field.widget, forms.CheckboxSelectMultiple):
+        del elementattributes["label"]
+        del elementattributes["required"]
 
         return hg.DIV(
             *[
                 Checkbox(
                     **{
-                        **kwargs,
+                        **elementattributes,
                         "label": widget.data["label"],
-                        "widgetattributes": extract_attrs(widget.data),
+                        "widgetattributes": {
+                            "name": widget.data["name"],
+                            "value": widget.data["value"],
+                            "checked": widget.data["selected"],
+                            **widget.data["attrs"],
+                            **widgetattributes,
+                        },
                     }
```

Comment:
I suggest you remove the outer-most dict, so you can write a bit more compact:

    Checkbox(
        **elementattributes,
        label=widget.data["label"],
        widgetattributes={
            "name": widget.data["name"],
            "value": widget.data["value"],
            "checked": widget.data["selected"],
            **widget.data["attrs"],
            **widgetattributes,
        }
    )